Automattic/jetpack
# Protect card: Add the Last scan time (with tooltip) based on plugin state.

This PR is the first PR in a series that will add/show various pieces of information on the Protect product card in My Jetpack. In this PR we will show:

1. if the Protect plugin is Not active, we will show the number of installed plugins and themes on the site. For Example: "12 Plugins & 3 Themes ⓘ" (with an info-icon popup).
2. If the Protect plugin is active, we will show the last scan time, For example: "Last scan: 17 hours ago".

See Design P2: p1HpG7-rFA-p2 (Figma link is within the post)

### Screenshots:

**Protect plugin _Not active_:**

![Screen Shot 2024-06-27 at 12 06 26](https://github.com/Automattic/jetpack/assets/11078128/4aed7300-2c73-4964-a1a4-918bdb29ef80)

![Screen Shot 2024-06-27 at 12 07 29](https://github.com/Automattic/jetpack/assets/11078128/f85b3a0e-ad6c-4d68-bd5e-15e4eebb9876)


**Protect plugin _Active_:**

![Screen Shot 2024-06-27 at 12 04 49](https://github.com/Automattic/jetpack/assets/11078128/ab81007f-faaf-40a3-8135-ce44ce301ea4)






**Note:** The "Scan", "Auto-Firewall", and "Logins-Blocked" sections are currently empty and not functional. Those will be added to in upcoming PR's.

## Proposed changes:

* Adds a 'themes' property to the `window.myJetpackInitialState` object.
* Adds a `protectStatus` property to the `window.myJetpackInitialState` object.
* Imports/uses the `protect-status` package.

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-aP6-p2

## Does this pull request change what data or activity we track or use?


 No

## Testing instructions:






- Run this PR locally (Docker dev environment) or from a Jurassic Ninja site using the Beta Tester plugin running this Jetpack branch.
- First make sure the Protect plugin is installed, but **Not** active.
- Go to My Jetpack and view the Protect product card.
- Verify you see the number of installed plugins and themes of the site on the Protect card (upper right corner).
- Verify you see an info icon (tooltip) and verify the info popup/tooltip works correctly and it matches the Figma design (linked in the Design post: p1HpG7-rFA-p2)
- Activate the Protect plugin. (Also make sure Jetpack is connected)
- Go back to My Jetpack and view the Protect card again.
- Verify you now see the "Last scan time:" with a proper value (in the upper right corner of the card)
    - On a brand new site, you may need to wait a minute for the initial first scan to complete, and then refresh the page to see the "Last scan time:" with a proper value. (although typically the first scan is very quick (just a couple seconds), in my experience).
- You can verify and compare the design with the Figma design (linked in the Design post: p1HpG7-rFA-p2)
